EIPS/eip-4519.md

```diff
@@ -0,0 +1,239 @@
+
+ ![Figure 3 : Flow chart of the token states with user and asset defined](../assets/eip-4519/images/Figure3.jpg)
+ 
+In order to complete a token transaction, a new owner must carry out a mutual authentication process, which is off-chain with the asset and on-chain with the token, by using their Ethereum accounts. Similarly, a new user must carry out a mutual authentication process. SmartNFTs define how the authentication processes start and finish. These authentication processes allow deriving fresh session cryptographic keys for secure communication between assets and owners, and between assets and users. Therefore, the trustworthiness of the assets can be traced even if new owners and users manage them. These processes are described below.
+ 
+```solidity
+pragma solidity ^0.8.0;
+
+ /// @title SmartNFT: Extension of ERC-721 Non-Fungible Token Standard. 
+///  Note: the ERC-165 identifier for this interface is 0x8a68abe3
+ interface SmartNFT is ERC721*,ERC165*{
+    /// @dev This emits when the NFT is assigned as utility of a new user.
+    ///  This event emits when the user of the token changes.
+    ///  (`_addressUser` == 0) when no user is assigned.
+    event UserAssigned(uint256 indexed tokenID, address indexed _addressUser);
+    
+    /// @dev This emits when user and asset finish mutual authentication process successfully.
+    ///  This event emits when both the user and the asset prove they share a secure communication channel.
+    event UserEngaged(uint256 indexed tokenID);
+    
+    /// @dev This emits when owner and asset finish mutual authentication process successfully.
+    ///  This event emits when both the owner and the asset prove they share a secure communication channel.
+    event OwnerEngaged(uint256 indexed tokenID);
+    
+    /// @dev This emits when it is checked that the timeout has expired.
+    ///  This event emits when the timestamp of the SmartNFT is not updated in timeout.
+    event TimeoutAlarm(uint256 indexed tokenID);
+
+    /// @notice This function defines how the NFT is assigned as utility of a new user (if user is defined).
+    /// @dev Only the owner of the SmartNFT can assign a user. If asset is defined, then the state of the token must be
+    /// "engagedWithOwner","waitingForUser" or "engagedWithUser" and this function changes the state of the token defined by "_tokenID" to
+    /// "waitingForUser". If asset is not defined, the state is set to "userAssigned". In both cases, this function sets the parameter 
+    /// "addressUser" to "_addressUser". 
+    /// @param _tokenId is the tokenID of the SmartNFT bound to the asset.
+    /// @param _addressUser is the address of the new user.
+    function setUser(uint256 _tokenId, address _addressUser) external; 
```

Comment:
Should the optional attributes (`user` and `asset`) be moved to separate interfaces?

Reply:
We don't  think so, because some functions work togethers, and have no sense in only one of these interfaces.